kolibri_explore_plugin/assets/src/views/SearchPage.vue

```diff
@@ -248,14 +255,23 @@
           return;
         }
 
-        kinds.forEach(k => searchChannels(this.$store, query, k));
+        this.progress = 0;
+        kinds.forEach(k => {
+          searchChannels(this.$store, query, k).then(() => {
+            this.progress += 100 / kinds.length;
+          });
```

Comment:
I really like your solution about doing the progress step by kind.

This is something very important to mention in the commit message. Please amend the commit explaining your implementation decision.